### DISCUSSION TITLE
Getting the best license_plate frame from mqtt events

### DISCUSSION THREAD

**Discussion by gadget-man** [blakeblackshear/frigate#9452] [General]
I'm trying to find the best way to ensure we send the best possible license plate snapshot to plate-recognizer for testing, to avoid repeatedly and unnecessarily sending multiple snapshots for processing.

Based on the mqtt response, I see there is already a `top_score`, which for objects you can compare the `before` and `after` to see whether there is a match / better candidate to process.

However, if you have Frigate+ and are detecting license_plates, there seems to be snapshot.attributes.l`{'label': 'license_plate', 'score': 0.51953125}`, but also `attributes': {'license_plate': 0.5625}`  as well as `current_attributes`. Please can you help me understand what the different scores mean, and which one is best to use as a comparison of `before` and `after` when deciding which is the best time to send the image for ANPR processing?

**Comment by NickM-27** [comment]
The mqtt docs explain the different fields https://deploy-preview-6262--frigate-docs.netlify.app/integrations/mqtt#frigateevents

```
    "attributes": {
      "face": 0.64
    }, // attributes with top score that have been identified on the object at any point
    "current_attributes": [] // detailed data about the current attributes in this frame
```

**Comment by gadget-man** [comment]
I did see that, but was a bit confused as to what `attributes with top score that have been identified on the object at any point` means, and the current docs don't provide any details of what the `snapshot.attributes` fields mean and how they differ from the above.

For our use case, if `"attributes":{"license_plate":X}` has a higher score in `after` then `before`, do I take it that that frame therefore has a better score? (sounds obvious, but I was confused by the reference to top score here and couldn't see the correlation between that and the equivalent score in `snapshot.attributes` for the license plate.).

**Reply by NickM-27** [reply]
`attributes` refers to attributes of that object. So if it is a car `attributes` will hold the `{'license_plate': top_score}` where `top_score` is the highest score that that specific car has had for its license plate attribute. I personally ignore this as it does not matter if you are looking for license plate, you want to look at `current_attributes` which holds information about attributes detected in the specific frame that just got updated

`snapshot` is a bit different, basically referring to the data around the currently saved best snapshot

**Reply by gadget-man** [reply]
Great thanks - I'll use `current_attributes` then
